justfile

```diff
@@ -0,0 +1,148 @@
+    test "${FORCE:-}" = "true" -o {{ src }} -nt {{ dst }} || exit 0
+    $BIN/pip-compile --allow-unsafe --output-file={{ dst }} {{ src }} {{ args }}
+
+
+_compile-prod src dst *args: virtualenv
+    #!/usr/bin/env bash
+    set -euo pipefail
+
+    # exit if src file is older than dst file (-nt = 'newer than', but we negate with || to avoid error exit code)
+    test "${FORCE:-}" = "true" -o {{ src }} -nt {{ dst }} || exit 0
+    $BIN/pip-compile --allow-unsafe --extras=drivers --output-file={{ dst }} {{ src }} {{ args }}
+
+
+# update requirements.prod.txt if requirements.prod.in has changed
+requirements-prod *args:
+    "{{ just_executable() }}" _compile-prod setup.py requirements.prod.txt {{ args }}
+
+
+# update requirements.dev.txt if requirements.dev.in has changed
+requirements-dev *args: requirements-prod
+    "{{ just_executable() }}" _compile-dev requirements.dev.in requirements.dev.txt {{ args }}
+
+
+# ensure prod requirements installed and up to date
+prodenv: requirements-prod
+    #!/usr/bin/env bash
+    set -euo pipefail
+
+    # exit if .txt file has not changed since we installed them (-nt == "newer than', but we negate with || to avoid error exit code)
+    test requirements.prod.txt -nt $VIRTUAL_ENV/.prod || exit 0
+
+    $PIP install -r requirements.prod.txt
+    touch $VIRTUAL_ENV/.prod
+
+
+# && dependencies are run after the recipe has run. Needs just>=0.9.9. This is
+# a killer feature over Makefiles.
+#
+# ensure dev requirements installed and up to date
+devenv: prodenv requirements-dev && install-precommit
+    #!/usr/bin/env bash
+    set -euo pipefail
+
+    # exit if .txt file has not changed since we installed them (-nt == "newer than', but we negate with || to avoid error exit code)
+    test requirements.dev.txt -nt $VIRTUAL_ENV/.dev || exit 0
+
+    $PIP install -r requirements.dev.txt
+    touch $VIRTUAL_ENV/.dev
+
+
+# ensure precommit is installed
+install-precommit:
+    #!/usr/bin/env bash
+    set -euo pipefail
+
+    BASE_DIR=$(git rev-parse --show-toplevel)
+    test -f $BASE_DIR/.git/hooks/pre-commit || $BIN/pre-commit install
+
+
+# upgrade dev or prod dependencies (specify package to upgrade single package, all by default)
+upgrade env package="": virtualenv
+    #!/usr/bin/env bash
+    set -euo pipefail
+
+    opts="--upgrade"
+    test -z "{{ package }}" || opts="--upgrade-package {{ package }}"
+    FORCE=true "{{ just_executable() }}" requirements-{{ env }} $opts
+
+
+# build the Docker image
+docker-build env="dev" *args="":
+    export BUILD_DATE=$(shell date +'%y-%m-%dT%H:%M:%S.%3NZ')
+    export REVISION=$(shell git rev-parse --short HEAD)
+    # We retrieve the version in GitHub Actions differently,
+    # so allow overriding this value.
+    export VERSION=${VERSION:-$(shell git describe --tags)}
+    export DOCKER_BUILDKIT=1
+    docker-compose build --pull {{ args }} {{ env }}
```

Comment:
I think you need to use `#!/usr/bin/env bash` for these env vars to be picked up by docker-compose - the just default is that each line is run in a separate shell, so they will be lost. e.g.

```
# justfile
myenv:
  export MYENVVAR=yeahyeahyeah
  echo $MYENVVAR
```

```bash
% just myenv
export MYENVVAR=yeahyeahyeah
echo $MYENVVAR
sh: MYENVVAR: unbound variable
error: Recipe `myenv` failed on line 25 with exit code 1
```